pkg/operator/operator_test.go

```diff
@@ -536,6 +616,58 @@ func TestSync(t *testing.T) {
 			},
 			expectErr: true,
 		},
+		{
+			// error: v1alpha1 VolumeSnapshot already exists
+			name:  "v1alpha1 VolumeSnapshot",
+			image: defaultImage,
+			initialObjects: testObjects{
+				csiSnapshotController: csiSnapshotController(),
+				crds:                  []*apiextv1beta1.CustomResourceDefinition{getAlphaCRD("VolumeSnapshot")},
+			},
+			expectedObjects: testObjects{
+				crds:                  []*apiextv1beta1.CustomResourceDefinition{getAlphaCRD("VolumeSnapshot")},
+				csiSnapshotController: csiSnapshotController(withTrueConditions(opv1.OperatorStatusTypeDegraded)),
+			},
+			expectErr: true,
+			reactors: testReactors{
+				crds: addCRDEstablishedRector,
+			},
+		},
+		{
+			// error: v1alpha1 VolumeSnapshotContent already exists
+			name:  "v1alpha1 VolumeSnapshotContent",
+			image: defaultImage,
+			initialObjects: testObjects{
+				csiSnapshotController: csiSnapshotController(),
+				crds:                  []*apiextv1beta1.CustomResourceDefinition{getAlphaCRD("VolumeSnapshotContent")},
+			},
+			expectedObjects: testObjects{
+				crds:                  []*apiextv1beta1.CustomResourceDefinition{getAlphaCRD("VolumeSnapshotContent")},
+				csiSnapshotController: csiSnapshotController(withTrueConditions(opv1.OperatorStatusTypeDegraded)),
+			},
+			expectErr: true,
+			reactors: testReactors{
+				crds: addCRDEstablishedRector,
+			},
+		},
+		{
+			// error: v1alpha1 VolumeSnapshotClass already exists
+			name:  "v1alpha1 VolumeSnapshotClass",
+			image: defaultImage,
+			initialObjects: testObjects{
+				csiSnapshotController: csiSnapshotController(),
+				crds:                  []*apiextv1beta1.CustomResourceDefinition{getAlphaCRD("VolumeSnapshotClass")},
+			},
+			expectedObjects: testObjects{
+				crds:                  []*apiextv1beta1.CustomResourceDefinition{getAlphaCRD("VolumeSnapshotClass")},
+				csiSnapshotController: csiSnapshotController(withTrueConditions(opv1.OperatorStatusTypeDegraded)),
+			},
+			expectErr: true,
+			reactors: testReactors{
+				crds: addCRDEstablishedRector,
+			},
+		},
```

Comment:
What about a case with v1beta1?

Reply:
v1beta1 is used in most previous cases, e.g. https://github.com/openshift/cluster-csi-snapshot-controller-operator/blob/b635111bb788b883653f8bce0d37c28457b10bf1/pkg/operator/operator_test.go#L400-L399